src/System.Web.WebPages/Helpers/AntiXsrf/AntiForgeryWorker.cs

```diff
@@ -104,7 +104,11 @@ public TagBuilder GetFormInputElement(HttpContextBase httpContext)
                 // Adding X-Frame-Options header to prevent ClickJacking. See
                 // http://tools.ietf.org/html/draft-ietf-websec-x-frame-options-10
                 // for more information.
-                httpContext.Response.AddHeader("X-Frame-Options", "SAMEORIGIN");
```

Comment:
Should this instead just be changed to:

```c#
httpContext.Response.Headers.Set("X-Frame-Options", "SAMEORIGIN");

```

This would make sure that anti-forgery gets the header it wants by overwriting any existing value.

Reply:
Thanks for getting back to me Eilon.  The "If"/Add check was to ensure that say if an administrator in IIS forced this header to a setting other than SAMEORIGIN (e.g. DENY or ALLOW FROM) it would allow them to.  If you always just set it to SAMEORIGIN you are not giving the admin the option to override it.

Reply:
I was just checking back in, does the above comment make sense?

Reply:
Ok this makes sense. I guess it's debatable which one should win over the other but I don't feel strongly about that, so this should be fine.